### ISSUE TITLE
[iq.com] Downloading a different dub audio tracks?

### ISSUE THREAD

**Issue by bombpersons** [yt-dlp/yt-dlp#3482]
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

### Question

Hi, I'm wondering if there currently is a supported way to specify the audio language to download? As you can see here: https://www.iq.com/play/it-s-a-drink-it-s-a-bomb-1985-19rwfk6z60?lang=en_us there is an option for a Cantonese or Mandarin audio track. Using yt-dlp I seem to only be able to download the Mandarin track. 

I'd thought that maybe if I set my language in my cookies differently it would affect the default dub, however that doesn't seem to have any effect.

**Comment by pukkandan** [comment]
Show output of `yt-dlp -F "URL"`

**Comment by bombpersons** [comment]
Unfortunately I only see the different resolutions as formats: 

`
$ ./yt-dlp.exe -F https://www.iq.com/play/it-s-a-drink-it-s-a-bomb-1985-19rwfk6z60?lang=en_us# --cookies cookies.txt                                            [iq.com] 19rwfk6z60: Downloading webpage
[iq.com] 19rwfk6z60: Downloading VIP data
[iq.com] 19rwfk6z60: Executing signature code
[iq.com] 19rwfk6z60: Downloading initial video format info
[iq.com] 19rwfk6z60: Downloading format data for 240P
[iq.com] 19rwfk6z60: Downloading format data for 1080P
[iq.com] 19rwfk6z60: Downloading format data for 720P
[iq.com] 19rwfk6z60: Downloading format data for 480P
[iq.com] 19rwfk6z60: Downloading format data for 360P
[info] Available formats for 19rwfk6z60:
ID  EXT RESOLUTION | PROTO  | VCODEC  ACODEC  MORE INFO
-------------------------------------------------------
100 mp4 384x216    | m3u8_n | unknown unknown 240P
200 mp4 640x360    | m3u8_n | unknown unknown 360P
300 mp4 896x504    | m3u8_n | unknown unknown 480P
500 mp4 1280x720   | m3u8_n | unknown unknown 720P
600 mp4 1920x1080  | m3u8_n | unknown unknown 1080P
`

**Comment by bombpersons** [comment]
Ah, thank's for the edit, was just fixing the formatting as you changed it, haha

**Comment by pukkandan** [comment]
@MinePlayersPE I assume this is the culprit?
https://github.com/yt-dlp/yt-dlp/blob/2d3b3feb7e69df0840d06fc1c8b27c5f26de054f/yt_dlp/extractor/iqiyi.py#L615

**Comment by MinePlayersPE** [comment]
> @MinePlayersPE Is this the culprit?
@pukkandan Yes, but extracting them is difficult as you'd need a second round of signature/cmd5x functions, along with the fact that they use a proprietary file format on top of AAC

**Comment by bombpersons** [comment]
Ahh that's unfortunate >.< Do you reckon there'd be an easy way for me to patch / hack the code to have it download the Cantonese audio and video instead? I can always run it twice and remux the audio together.

**Comment by bombpersons** [comment]
So I had a shot at this myself: https://github.com/bombpersons/yt-dlp

I managed to figure out how to get the audio fragment urls, needed some more of that magic cmd5x function. 

The audio is actually OK with a bit of modification, just the first fragment (which I assume is the header) is gzipped. With the rest concatenated it works!

My code is a mess but works for just downloading the audio seperately at least. Might have some time to clean it up. Feel free to give me some pointers if you like or work on it yourself =)
